### PR TITLE
ddl: migrate test-infra to testify for placement_policy_test

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: ref #27180

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/placement_policy_test.go [hunk 1]
```diff
@@ -34,73 +34,63 @@ import (
 	"github.com/pingcap/tidb/parser/model"
 	"github.com/pingcap/tidb/sessionctx"
 	"github.com/pingcap/tidb/store/gcworker"
-	"github.com/pingcap/tidb/util/testkit"
+	"github.com/pingcap/tidb/testkit"
 	"github.com/pingcap/tidb/util/testutil"
```
Comment (tisonkun) [inline ddl/placement_policy_test.go:38]: You can replace `testutil.RowsWithSep` to `testkit.RowsWithSep` (new testkit has implemented this method) and remove this dep. I think `util/testutil` is also dep to be deprecated.

Comment (hawkingrei) [reply ddl/placement_policy_test.go]: have removed.

File: ddl/placement_policy_test.go [hunk 2]
```diff
@@ -192,19 +183,22 @@ func (s *testDBSuite6) TestPlacementPolicy(c *C) {
...
-	c.Assert(0, Equals, len(bundles))
+	require.NoError(t, err)
+	require.Equal(t, len(bundles), 0)
 
 	tk.MustExec("drop placement policy x")
 	tk.MustGetErrCode("drop placement policy x", mysql.ErrPlacementPolicyNotExists)
-	tk.MustExec("drop placement policy if exists x")
-	tk.MustQuery("show warnings").Check(testkit.Rows("Note 8239 Unknown placement policy 'x'"))
+	tk.MustGetErrCode("drop placement policy if exists x", mysql.ErrPlacementPolicyNotExists)
+	//tk.MustExec("drop placement policy if exists x")
```
Comment (tisonkun) [inline ddl/placement_policy_test.go:192]: Do you know why the line failed?

cc @morgo IIRC you implement placement rule in SQL, do you have ideas here?

Comment (hawkingrei) [reply ddl/placement_policy_test.go:192]: @tisonkun I have reported this problem to @lcwangchao. He is surveying this problem.

Comment (hawkingrei) [reply ddl/placement_policy_test.go]: Have fixed. Although we don't know why it succeeds to run in the past.

**Review by tisonkun — APPROVED** [review]

File: ddl/placement_policy_test.go [hunk 3]
```diff
@@ -190,10 +180,10 @@ func (s *testDBSuite6) TestPlacementPolicy(c *C) {
 		"PRIMARY_REGION=\"cn-east-1\" " +
 		"REGIONS=\"cn-east-1,cn-east-2\" ")
 	tk.MustQuery("show warnings").Check(testkit.Rows("Note 8238 Placement policy 'X' already exists"))
-
+	dom.DDL().SetHook(originalHook)
```
Comment (lcwangchao) [inline ddl/placement_policy_test.go:183]: I think it's better to remove this line and init custom hook like this:

```
hook := &ddl.TestDDLCallback{Do: dom}
```

In this way, the custom hook can help reload schema automatically

Comment (hawkingrei) [reply ddl/placement_policy_test.go]: Fixed.

**Review by xiongjiwei — APPROVED** [review]

**Pull request merged**